worlds/yachtdice/__init__.py

```diff
@@ -0,0 +1,404 @@
+            k=self.number_of_locations - already_items
+        )
+
+        #we are done adding items. Now because of the last step, number of items should be number of locations
+        already_items = len(self.itempool) + self.extra_plando_items + 1
+        if already_items != self.number_of_locations:
+            raise Exception(f"[Yacht Dice] Number in self.itempool is not number of locations "
+                f"{already_items} {self.number_of_locations}.")
+        
+        #add precollected items using push_precollected. Items in self.itempool get created in create_items
+        for item in self.precollected:
+            self.multiworld.push_precollected(self.create_item(item))
+
+    def create_items(self):
+        #convert strings to actual items
+        itempool_created = [item for item in map(lambda name: self.create_item(name), self.itempool)]
+
+        #and add them to the itempool
+        for item in itempool_created:
+            self.multiworld.itempool += [item]    
+
+    def create_regions(self):
+        #call the ini_locations function, that generates locations based on the inputs.
+        location_table, goal_index = ini_locations(self.goal_score, self.max_score, self.number_of_locations, 
+                                                        self.options.game_difficulty.value)
+
+        #simple menu-board construction
+        menu = Region("Menu", self.player, self.multiworld)
+        board = Region("Board", self.player, self.multiworld)
+
+        #add locations to board, one for every location in the location_table
+        board.locations = [YachtDiceLocation(self.player, loc_name, loc_data.score, loc_data.id, board)
+                            for loc_name, loc_data in location_table.items() if loc_data.region == board.name]
+        
+        #which index of all locations should have the Victory item.
+        
+        # Add the victory item to the correct location. 
+        # The website declares that the game is complete when the victory item is obtained.
+        board.locations[goal_index].place_locked_item(self.create_item("Victory"))
+              
+        #these will be slot_data input
+        self.goal_score = board.locations[goal_index].yacht_dice_score
+        self.max_score = board.locations[-1].yacht_dice_score
+        
+        #add the regions
+        connection = Entrance(self.player, "New Board", menu)
+        menu.exits.append(connection)
+        connection.connect(board)
+        self.multiworld.regions += [menu, board]
+        
+    def set_rules(self):
+        #set rules per location, and add the rule for beating the game
+        set_yacht_rules(self.multiworld, self.player, self.options)
+        set_yacht_completion_rules(self.multiworld, self.player)
+        
+    def pre_fill(self):
+        #in the pre_fill, make sure one dice and one roll is early, so that you will have 2 dice and 2 rolls soon
+        self.multiworld.early_items[self.player]["Dice"] = 1
```

Comment:
Early items should be added in `generate_early`

Reply:
OK, thanks, I've moved it to the end of `generate_early`